libcalico-go/lib/backend/k8s/conversion/workload_endpoint_default.go

```diff
@@ -253,6 +253,14 @@ func (wc defaultWorkloadEndpointConverter) podToDefaultWorkloadEndpoint(pod *kap
 		AllowSpoofedSourcePrefixes: requestedSourcePrefixes,
 	}
 
+	for key := range pod.Annotations {
```

Comment:
I think we can actually do this more efficiently, since we're matching on the exact key.

e.g.,

```go
if v, ok := pod.Annotations["k8s.v1.cni.cncf.io/network-status"]; ok {
  	if wep.Annotations == nil {
		wep.Annotations = make(map[string]string)
	}
        wep.Annotations["k8s.v1.cni.cncf.io/network-status"] = v
}
```

^ That way we don't need to loop through the annotations.